extension/icu/icu-datefunc.cpp

```diff
@@ -73,6 +73,9 @@ unique_ptr<FunctionData> ICUDateFunc::Bind(ClientContext &context, ScalarFunctio
 
 void ICUDateFunc::SetTimeZone(icu::Calendar *calendar, const string_t &tz_id) {
 	auto tz = icu_66::TimeZone::createTimeZone(icu::UnicodeString::fromUTF8(icu::StringPiece(tz_id.GetString())));
+	if (*tz == icu::TimeZone::getUnknown()) {
+		throw NotImplementedException("Unknown TimeZone '%s'", tz_id.GetString());
```

Comment:
```suggestion
		throw InvalidInputException("Unknown TimeZone '%s'", tz_id.GetString());
```
Maybe somewhat better.

Reply:
Ah this is the same exception we are using for `SET TimeZone=` Hard to say whether it is not implemented (they gave us a legal TZ that we don't support) or invalid (they gave us a TZ that doesn't exist anywhere.)

Reply:
right, cool then